tests/integration/nm/testlib.py

```diff
@@ -36,10 +37,12 @@ def mainloop():
         raise MainloopTestError(mloop.error)
 
 
+@nmclient_context
```

Comment:
This looks strange. It decorates the decorator. But the decorator is called at `import` time once, do something is fishy here.